proposals/6845-add-a-parameter-for-scroll-speed.md

```diff
@@ -0,0 +1,62 @@
+---
+title: Add a parameter for scroll speed
+author:
+- Andrey Fedorovich
+date: 30.01.2024
+---
+
+# RFC6845: Add a parameter for scroll speed
+
+## Summary
+
+To manipulate the scroll animation speed, developers now need to create the animation themselves instead of using it out of the box
+
+## Basic example
+
+The possible way to implement that functionality could be:
+
+```javascript
+scrollToIndex: (params: {
+  index: number;
+  animated?: boolean;
+  viewOffset?: number;
+  viewPosition?: number;
+  speed?: number; // new value 
+});
+```
+
+## Motivation
+
+Usage of FlatList scrollToIndex function with an animation speed parameter improves the developers' experience
+
+## Detailed design
```

Comment:
I think on iOS this can be achieved pretty easily using `animateWithDuration`. 

I've modified the code of `RCTScrollView` with this snippet and it works. I think providing a _native_ way of handling this will be really nice addition to the Framework: 
```objc
[UIView animateWithDuration:10.0f delay:0 options:UIViewAnimationOptionCurveLinear animations:^{
      self->_scrollView.contentOffset = offset;
    } completion:NULL];
```